packages/lesswrong/integrationTests/integrationTestSetup.ts

```diff
@@ -73,7 +73,14 @@ afterEach(async () => {
 });
 
 afterAll(async () => {
-  await waitUntilCallbacksFinished();
+  // Enforce a wait of at least a 1 second before closing the database connection. To mitigate these potential issues:
+  // - There may be some dead time between callbacks where waitUntilCallbacksFinished resolves, but actually another callback is about to run
+  // - Some async functions may not be caught by waitUntilCallbacksFinished at all
+  for (let i = 0; i < 10; i++) {
+    await waitUntilCallbacksFinished();
+    await new Promise((resolve) => setTimeout(resolve, 100));
+  }
```

Comment:
I tried fairly hard to avoid doing this, but making waitUntilCallbacksFinished was actually very difficult so I've added this hack to hopefully prevent most failures. Compared to the total time the tests take to run this isn't too bad

Reply:
Note: unfortunately test suites do not run concurrently, so this adds about 16s overall, which is quite a lot

Reply:
Yeah - this is pretty sad. It looks like we currently have about 50 places in the callbacks directory where we `void` promises. We probably need to change most/all of these to `await` if we want `waitUntilCallbacksFinished` to be reliable.